### PR TITLE
Rename to support non text element references

### PR DESCRIPTION
**Pull request by samvaity**
Property renames:
labelText -> labelData
valueText -> valueData
textContent -> fieldElements

Model renames:
FieldText -> FieldData
FormContent -> FormElement

Parameter rename:
includeTextContent -> includeFieldElements

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md [hunk 1]
```diff
@@ -2,19 +2,23 @@
 
 ## 1.0.0-beta.4 (Unreleased)
 ### Breaking Changes
-- Rename properties `requestedOn` to `trainingStartedOn` and `completedOn` to `trainingCompletedOn` on model
+- Renamed property `includeTextContent` to `includeFieldElement` on `RecognizeOptions` model.
+- Methods returning `textContent` have been renamed to `getFieldElements()` on `FieldData` and `FormTableCell`
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md:6]: should this be renamed to `fieldElements`?

File: sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md [hunk 2]
```diff
@@ -2,19 +2,23 @@
 
 ## 1.0.0-beta.4 (Unreleased)
 ### Breaking Changes
-- Rename properties `requestedOn` to `trainingStartedOn` and `completedOn` to `trainingCompletedOn` on model
+- Renamed property `includeTextContent` to `includeFieldElement` on `RecognizeOptions` model.
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md:5]: `includeFieldElements`? I think this should be plural

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/FormField.java [hunk 3]
```diff
@@ -34,24 +34,24 @@
...
      */
-    private final FieldText valueText;
+    private final FieldData valueData;
 
     /**
      * Constructs a FormField object.
      *
      * @param confidence The confidence of the recognized field.
-     * @param labelText The label text value for the field.
+     * @param labelData The label text value for the field.
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/FormField.java:43]: docstring should be updated.
I think we say:
"Contains the text, bounding box, and field elements for the field label."

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/FormField.java [hunk 4]
```diff
@@ -11,47 +11,28 @@
...
      *
      * @param confidence The confidence of the recognized field.
-     * @param labelText The label text value for the field.
-     * @param name The name the field.
-     * @param fieldValue The value of the field.
-     * @param valueText The label value text for the field.
+     * @param labelData The text, bounding box, and field elements for the field label.
+     * @param name The name the field or label.
+     * @param fieldValue The value of the recognized field.
+     * @param valueData The the text, bounding box, and field elements for the field value.
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/FormField.java:27]: typo, duplicate "The the text, ..."

Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/FormField.java:27]:
```suggestion
     * @param valueData The text, bounding box, and field elements for the field value.
```

File: sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/FormRecognizerClientJavaDocCodeSnippets.java [hunk 5]
```diff
@@ -119,7 +119,7 @@ public void beginRecognizeCustomFormsWithOptions() throws IOException {
 
         formRecognizerClient.beginRecognizeCustomForms(new RecognizeCustomFormsOptions(targetStream,
             form.length(), modelId).setFormContentType(FormContentType.IMAGE_JPEG)
-            .setIncludeTextContent(includeTextContent).setPollInterval(Duration.ofSeconds(5))).getFinalResult()
+            .setIncludeFieldElements(includeTextContent).setPollInterval(Duration.ofSeconds(5))).getFinalResult()
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/FormRecognizerClientJavaDocCodeSnippets.java:122]: nit: might need to consider local var `includeTextContent` to 'includeFieldElements' for the renaming as well.

File: sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md [hunk 6]
```diff
@@ -2,19 +2,22 @@
...
-- Add `RecognizeOptions` and `RecognizeCustomFormOptions` to pass configurable options when using recognize APIs on FormRecognizerClient.
-- Change `submodels` property on `CustomFormModel` to return a `List` instead of `IterableStream`
-- Rename `fieldMap` property to `fields` on `CustomFormSubmodel` model
-- Rename `elements` property on model `FormTableCell` to `textContent`
-- Rename `includeTextDetails` references in parameter and model properties to `includeTextContent`
-- Remove `TextContentType` model and use `instanceOf` to detect the FormContent type
+- Changed `CopyAuthorization.getExpiresOn()` to return a `OffsetDateTime` instead of a `long` value
+- Added `RecognizeOptions` and `RecognizeCustomFormOptions` to pass configurable options when using recognize APIs on FormRecognizerClient.
+- Changed `submodels` property on `CustomFormModel` to return a `List` instead of `IterableStream`
+- Renamed `fieldMap` property to `fields` on `CustomFormSubmodel` model
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md:14]: nit: a better look if group rename changelog all together.

File: sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/FormRecognizerClientJavaDocCodeSnippets.java (hunk 5 above)
Comment (samvaity) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/FormRecognizerClientJavaDocCodeSnippets.java]: updated!!

File: sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/GetBoundingBoxes.java [hunk 7]
```diff
@@ -68,11 +68,11 @@ public static void main(String[] args) {
                     System.out.printf("Table %d%n", i2);
                     formTable.getCells().forEach(formTableCell -> {
                         System.out.printf("Cell text %s has following words: %n", formTableCell.getText());
-                        // textContent only exists if you set includeTextContent to True in your
+                        // textContent only exists if you set includeFieldElements to True in your
```
Comment (maririos) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/GetBoundingBoxes.java:71]: this should be `fieldElements`

File: sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/GetBoundingBoxesAsync.java [hunk 8]
```diff
@@ -82,11 +82,11 @@ public static void main(String[] args) {
                         System.out.printf("Table %d%n", i2);
                         formTable.getCells().forEach(formTableCell -> {
                             System.out.printf("Cell text %s has following words: %n", formTableCell.getText());
-                            // textContent only exists if you set includeTextContent to True in your
+                            // textContent only exists if you set includeFieldElements to True in your
```
Comment (maririos) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/GetBoundingBoxesAsync.java:85]: same here

File: sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClientTest.java [hunk 9]
```diff
@@ -219,39 +219,39 @@ public void recognizeReceiptInvalidSourceUrl(HttpClient httpClient, FormRecogniz
     }
 
     /**
-     * Verifies receipt data for a document using source as file url and include content when includeTextContent is
+     * Verifies receipt data for a document using source as file url and include content when includeFieldElements is
      * true.
      */
     @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
     @MethodSource("com.azure.ai.formrecognizer.TestUtils#getTestParameters")
     public void recognizeReceiptFromUrlTextContent(HttpClient httpClient,
```
Comment (maririos) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClientTest.java:227]: should this test be renamed too? same the description of it?

Comment (maririos) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClientTest.java:227]: and same for other tests

File: sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/FormRecognizerClientJavaDocCodeSnippets.java [hunk 10]
```diff
@@ -112,14 +112,14 @@ public void beginRecognizeCustomFormsWithOptions() throws IOException {
...
 
         byte[] fileContent = Files.readAllBytes(form.toPath());
         InputStream targetStream = new ByteArrayInputStream(fileContent);
 
         formRecognizerClient.beginRecognizeCustomForms(new RecognizeCustomFormsOptions(targetStream,
             form.length(), modelId).setFormContentType(FormContentType.IMAGE_JPEG)
-            .setIncludeTextContent(includeTextContent).setPollInterval(Duration.ofSeconds(5))).getFinalResult()
+            .setIncludeFieldElements(includeFieldElements).setPollInterval(Duration.ofSeconds(5))).getFinalResult()
             .forEach(recognizedForm -> recognizedForm.getFields().entrySet().forEach(entry -> {
                 String fieldText = entry.getKey();
```
Comment (maririos) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/FormRecognizerClientJavaDocCodeSnippets.java:124]: should we change this to fieldData?

File: sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/GetBoundingBoxes.java [hunk 11]
```diff
@@ -50,7 +50,7 @@ public static void main(String[] args) {
             recognizedForm.getFields().forEach((fieldText, fieldValue) -> System.out.printf("Field %s has value %s "
```
Comment (maririos) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/GetBoundingBoxes.java:50]: same here?

File: sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/GetBoundingBoxes.java [hunk 12]
```diff
@@ -68,11 +68,11 @@ public static void main(String[] args) {
                     System.out.printf("Table %d%n", i2);
                     formTable.getCells().forEach(formTableCell -> {
                         System.out.printf("Cell text %s has following words: %n", formTableCell.getText());
-                        // textContent only exists if you set includeTextContent to True in your
+                        // textContent only exists if you set includeFieldElements to True in your
                         // call to beginRecognizeCustomFormsFromUrl
                         // It is also a list of FormWords and FormLines, but in this example, we only deal with
                         // FormWords
-                        formTableCell.getTextContent().forEach(formContent -> {
+                        formTableCell.getFieldElements().forEach(formContent -> {
```
Comment (maririos) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/GetBoundingBoxes.java:75]: formelement?

File: sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/AdvancedDiffLabeledUnlabeledData.java [hunk 13]
```diff
@@ -97,30 +97,30 @@ public static void main(String[] args) throws IOException {
         // `value` will contain the typed field value and `valueText` will contain information about the field value
```
Comment (maririos) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/AdvancedDiffLabeledUnlabeledData.java:97]: this needs to change too

**Pull request merged**